azavea/echo-locator
# Add neighborhood images to database

## Overview

This PR adds neighborhood images to the database for easier saving and editing. It also adds image importing to the `import_neighborhood_data` script (previously named `import_neighborhood_bounds`) and removes the now unnecessary `neighborhoods.json` and `neighborhood_bounds.json` files.


### Checklist


### Demo

The images as they are displayed in the frontend.

<img width="429" alt="Screen Shot 2022-06-21 at 9 07 56 AM" src="https://user-images.githubusercontent.com/77936689/174806786-7bf26f19-fd4d-4ab3-8b8e-e53adf207f0f.png">

<img width="410" alt="Screen Shot 2022-06-21 at 9 08 14 AM" src="https://user-images.githubusercontent.com/77936689/174806798-119f5388-8a96-4ee0-9f71-a2c36ba26641.png">

Users can edit and replace the images from the admin.

<img width="955" alt="Screen Shot 2022-06-21 at 10 22 30 AM" src="https://user-images.githubusercontent.com/77936689/174823348-b2b0eae2-e3dd-4fe8-8540-0df5f6c6c34d.png">


### Notes

This PR saves the images to S3 via a custom storage backend. This set up required a few additions, including:
* adding `aws_s3_photo_bucket` environment variable to Terraform and the `terraform.tfvars` file in S3
* adding `aws_s3_photo_bucket` environment variable to `.env` and telling Django to use `.env` for local development. This change to `.env` has been updated in LastPass
* In order to guarantee local development has the required permissions without breaking STRTA, `AWS_PROFILE=echo-locator` has been added to `./scripts/server`

This PR adds two new fields for each image, an image field that allows you to update the file, and an image preview field. As a result of this, there are now a total of 9 fields per image: the original Wikipedia link, the Wikipedia thumbnail link, the image field with the link to S3, the image preview field, the image license, license URL, description, artist, and artist username. The first three of these fields seem especially redundant and all fields could probably be organized differently, but I have left them as is until the client tells us if they would prefer something else. 

Due to budget, I did not add any validation to picture size, only help text to instruct users as to what size and ratio the images will be displayed at. It seems to do fine with images that are over-sized but not with pictures that are portrait rather than landscape. I also did not add cropping to the images; this was for two reasons: 1) budget and 2) the images are downloaded and displayed at such a small size that cropping would have been difficult, anyway.


## Testing Instructions

 * Run `./scripts/update` to apply the image field migration
 * Run `AWS_PROFILE=echo-locator ./scripts/manage import_neighborhood_data s3://echolocator-production-data-us-east-1/neighborhoods.json s3://echolocator-production-data-us-east-1/neighborhood_bounds.json s3://echo-locator-production-site-us-east-1/assets/neighborhoods/` to import the images from S3
 * Verify that the images display as they should in the admin
 * Verify that the images display as they should both on the neighborhood card in the dock and in the neighborhood detail view


Resolves #494 

EDIT: Changed length of image field help text to fix spacing issues.
